src/IO/CMakeLists.txt

```diff
@@ -24,9 +24,12 @@ add_spectre_library(${LIBRARY} ${LIBRARY_SOURCES})
 
 target_link_libraries(
   ${LIBRARY}
-  INTERFACE DataStructures
-  INTERFACE ErrorHandling
-  INTERFACE Utilities
+  PRIVATE
```

Comment:
I'm still pretty sure interface is the right call here...

Reply:
I made all of these libraries `PUBLIC` because I think they are all used both in source and header files. Do you agree this is the right thing to do?